src/Fixer/FunctionNotation/MethodArgumentSpaceFixer.php

```diff
@@ -127,8 +156,104 @@ private function fixFunction(Tokens $tokens, $startFunctionIndex)
+                [[T_WHITESPACE]]
+            );
+            $searchIndex = $prevWhitespaceTokenIndex;
+        } while ($prevWhitespaceTokenIndex
+            && false === strpos($tokens[$prevWhitespaceTokenIndex]->getContent(), "\n")
+        );
+        $existingIndentation = $prevWhitespaceTokenIndex
+            ? ltrim($tokens[$prevWhitespaceTokenIndex]->getContent(), "\n\r")
+            : '';
+
+        $indentation = $existingIndentation.$this->whitespacesConfig->getIndent();
+        $endFunctionIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startFunctionIndex);
+        if (!$this->isNewline($tokens[$endFunctionIndex - 1])) {
+            $tokens->insertAt(
+                $endFunctionIndex,
+                new Token([
+                    T_WHITESPACE,
+                    $this->whitespacesConfig->getLineEnding().$existingIndentation,
+                ])
+            );
+            ++$endFunctionIndex;
+        }
+
+        for ($index = $endFunctionIndex - 1; $index > $startFunctionIndex; --$index) {
+            $token = $tokens[$index];
+
+            // skip nested method calls and arrays
+            if ($token->equals(')')) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index, false);
+                continue;
+            }
+
+            // skip nested arrays
+            if ($token->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_CLOSE)) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $index, false);
+                continue;
+            }
+
+            if ($token->equals(',')) {
+                $this->fixNewline($tokens, $index, $indentation);
+            }
+        }
+        $this->fixNewLine($tokens, $startFunctionIndex, $indentation, false);
+    }
+
+    /**
+     * Method to insert newline after comma or opening parenthesis.
+     *
+     * @param Tokens $tokens
+     * @param int    $index       index of a comma
+     * @param string $indentation the indentation that should be used
+     * @param bool   $override    whether to override the existing character or not
+     */
+    private function fixNewline(Tokens $tokens, $index, $indentation, $override = true)
+    {
+        if ($this->isNewline($tokens[$index + 1]) || $tokens[$index + 1]->isComment()) {
+            return;
+        }
+        if ($tokens[$index + 2]->isComment()) {
+            $nextMeaningfulTokenIndex = $tokens->getNextMeaningfulToken($index + 2);
+            if (!$this->isNewLine($tokens[$nextMeaningfulTokenIndex - 1])) {
+                $tokens->insertAt(
+                    $nextMeaningfulTokenIndex,
+                    new Token([
+                        T_WHITESPACE,
+                        $this->whitespacesConfig->getLineEnding().$indentation,
+                    ])
+                );
+            }
+
+            return;
+        }
+
+        call_user_func(
+            [$tokens, $override ? 'overrideAt' : 'insertAt'],
```

Comment:
I renamed it because `ensureWhitespaceAtIndex` does many things that I don't do there.